src/ocean/math/BinaryHistogram.d

```diff
@@ -0,0 +1,384 @@
+/*******************************************************************************
+
+    An integer, binary logarithmic statistics counter
+
+    Collects the following statistical information:
+    - a logarithmic byte throughput histogram with bins from ≥1B to <1<<MaxPow2
+      (see template documentation, below), one bin per power of two, plus one
+      bin for each 0B and ≥1<<MaxPow2,
+    - the total number of transactions and the aggregated total amount.
+
+    To reset all counters to zero use `BinaryHistogram.init`.
+
+    Copyright:
+        Copyright (c) 2017-2018 sociomantic labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE.txt for details.
+
+*******************************************************************************/
+
+module ocean.math.BinaryHistogram;
+
+import ocean.transition;
+import ocean.core.Verify;
+
+/*******************************************************************************
+
+    An integer, binary logarithmic statistics counter
+
+    Params:
+        MaxPow2 = the maximum power of two that is tracked in the lower bins of
+            the counter. Any values higher than 1<<MaxPow2 will be aggregated in
+            the upper bin
```

Comment:
State the allowed maximum for `MaxPow2`, imposed by the limited number of order-of-maginute suffixes (the highest being `Ei`).

Reply:
Maybe add a unit test for `MaxPow2 == 0`.